apps/token/src/routes/governance/components/proposals-list-item/proposals-list-item-details.spec.tsx

```diff
@@ -0,0 +1,374 @@
+import { DATE_FORMAT_DETAILED } from '../../../../lib/date-formats';
+import {
+  mockPubkey,
+  mockWalletContext,
+  networkParamsQueryMock,
+  amendToDate,
+  lastWeek,
+  nextWeek,
+} from '../../test-helpers/mocks';
+import type { Proposals_proposals } from '../../proposals/__generated__/Proposals';
+
+const oneDay = 60 * 60 * 24;
+const oneHour = 60 * 60;
+const oneMinute = 60;
+
+const renderComponent = (
+  proposal: Proposals_proposals,
+  mock = networkParamsQueryMock
+) => (
+  <Router>
+    <MockedProvider mocks={[mock]}>
+      <AppStateProvider>
+        <VegaWalletContext.Provider value={mockWalletContext}>
+          <ProposalsListItemDetails proposal={proposal} />
+        </VegaWalletContext.Provider>
+      </AppStateProvider>
+    </MockedProvider>
+  </Router>
+);
+
+describe('Proposals list item details', () => {
+  it('Renders proposal state: Enacted', () => {
+    render(
+      renderComponent(
+        generateProposal({
+          state: ProposalState.Enacted,
+          terms: {
+            enactmentDatetime: lastWeek.toString(),
+          },
+        })
+      )
+    );
+    expect(screen.getByTestId('proposal-status')).toHaveTextContent('Enacted');
+    expect(screen.getByTestId('vote-details')).toHaveTextContent(
+      format(lastWeek, DATE_FORMAT_DETAILED)
+    );
+  });
+
+  it('Renders proposal state: Passed', () => {
+    render(
+      renderComponent(
+        generateProposal({
+          state: ProposalState.Passed,
+          terms: {
+            closingDatetime: lastWeek.toString(),
+            enactmentDatetime: nextWeek.toString(),
+          },
+        })
+      )
+    );
+    expect(screen.getByTestId('proposal-status')).toHaveTextContent('Passed');
+    expect(screen.getByTestId('vote-details')).toHaveTextContent(
+      `Enacts on ${format(nextWeek, DATE_FORMAT_DETAILED)}`
+    );
+  });
+
+  it('Renders proposal state: Waiting for node vote', () => {
+    render(
+      renderComponent(
+        generateProposal({
+          state: ProposalState.WaitingForNodeVote,
+          terms: {
+            enactmentDatetime: nextWeek.toString(),
+          },
+        })
+      )
+    );
+    expect(screen.getByTestId('proposal-status')).toHaveTextContent(
+      'Waiting for node vote'
+    );
+    expect(screen.getByTestId('vote-details')).toHaveTextContent(
+      `Enacts on ${format(nextWeek, DATE_FORMAT_DETAILED)}`
+    );
+  });
+
+  it('Renders proposal state: Open - 5 minutes left to vote', () => {
+    const fiveMinutes = amendToDate(oneMinute * 5);
+    render(
+      renderComponent(
+        generateProposal({
+          state: ProposalState.Open,
+          terms: {
+            closingDatetime: fiveMinutes.toString(),
+          },
+        })
+      )
+    );
+    expect(screen.getByTestId('proposal-status')).toHaveTextContent('Open');
+    expect(screen.getByTestId('vote-details')).toHaveTextContent(
+      '5 minutes left to vote'
```

Comment:
chore (non-blocking): we may need to use mock timers, this test could potentially time out or be flakey is it takes a long time to render or similar. Let's keep an eye on it.